File: Chapter9/Java/Writing Methods that Require a Single Parameter in Java/SumAndProduct.java
// SumAndProduct.java - This program computes sums and products. 
// Input:  Interactive.
// Output:  Computed sum and product. 

import java.util.Scanner;

class SumAndProduct {

    static int sums(int x) {
        int ans = 0;
        for (int i = 1; i <= x; i++) {
            ans += i;
        }
        return ans;
    }

    static int products(int x) {
        int ans = 1;
        for (int i = 1; i <= x; i++) {
            ans *= i;
        }
        return ans;
    }

    public static void main(String args[]) {
        Scanner s = new Scanner(System.in);
        int number;
        String numberString;
        System.out.println("Enter a positive integer or 0 to quit: ");
        numberString = s.nextLine();
        number = Integer.parseInt(numberString);
        while (number != 0) {
            // call sums() method here
            int sum = sums(number);
            int prod = products(number);
            System.out.println("Sum = " + sum + " , Prod = " + prod);
            // call products() method here
            System.out.println("Enter a positive integer or 0 to quit: ");
            numberString = s.nextLine();
            number = Integer.parseInt(numberString);
        }
        // System.exit(0);
    }
    // End of main() method.
    // Write sums() method here.
    // Write products() method here.
    // }
    //
}

File: Chapter9/Python/Writing Functions that Require a Single Parameter in Python/SumAndProduct.py
"""
SumAndProduct.py - This program computes sums and products.
Input:  Interactive.
Output:  Computed sum and product.
"""


# Write sums() function here.
def sums(n):
    result = 0
    for i in range(1, n+1):
        result += i
    return result

# Write products() function here.
def products(n):
    result = 1
    for i in range(1, n + 1):
        result *= i
    return result

numberString = input("Enter a positive integer or 0 to quit: ")
number = int(numberString)

while number != 0:
    # Call sums() function here.
    print(sums(number))
    # Call products() function here.
    print(products(number))

    numberString = input("Enter a positive integer or 0 to quit: ")
    number = int(numberString)
